src/main/java/reposense/parser/CsvParser.java

```diff
@@ -51,105 +51,107 @@ public CsvParser(Path csvFilePath) throws IOException {
-        if (isElementOverridingStandaloneConfig(elements, position)) {
-            elements[position] = elements[position].replaceFirst(OVERRIDE_KEYWORD, "");
-        }
+    protected String getOrDefault(final CSVRecord record, int colNum, String defaultValue) {
+        return record.get(colNum).isEmpty() ? defaultValue : get(record, colNum);
     }
 
     /**
-     * Gets the value of {@code position} in {@code elements}.
-     * Returns the value of {@code position} if it is in {@code element} and not empty.
-     * Otherwise returns an empty string.
+     * Returns the value of {@code record} at {@code colNum} as a {@code List},
+     * delimited by {@code COLUMN_VALUES_SEPARATOR} if it is in {@code record} and not empty, or
+     * returns an empty {@code List} otherwise.
      */
-    protected String getValueInElement(final String[] elements, int position) {
-        return (containsValueAtPosition(elements, position)) ? elements[position] : "";
+    protected List<String> getAsList(final CSVRecord record, int colNum) {
+        if (record.get(colNum).isEmpty()) {
+            return Collections.emptyList();
+        }
+        return Arrays.stream(record.get(colNum).split(COLUMN_VALUES_SEPARATOR))
+                .map(String::trim)
+                .collect(Collectors.toList());
     }
 
     /**
-     * Gets the value of {@code position} in {@code elements}.
-     * Returns the value of {@code position} if it is in {@code element} and not empty.
-     * Otherwise returns the {@code defaultValue}.
+     * Returns the values in {@code record} as a list with the {@link CsvParser#OVERRIDE_KEYWORD} prefix removed.
+     * Returns an empty list if {@code record} at {@code colNum} is empty.
      */
-    protected String getValueInElement(final String[] elements, int position, String defaultValue) {
-        return (containsValueAtPosition(elements, position)) ? elements[position] : defaultValue;
-    }
+    protected List<String> getAsListWithoutOverridePrefix(final CSVRecord record, int colNum) {
+        String rawValue = (isElementOverridingStandaloneConfig(record, colNum))
+                ? record.get(colNum).replaceFirst(OVERRIDE_KEYWORD, "")
+                : record.get(colNum);
 
-    /**
-     * Gets the value of {@code position} in {@code elements}.
-     * Returns the value of {@code element} at {@code position} as a {@code List},
-     * delimited by {@code COLUMN_VALUES_SEPARATOR} if it is in {@code element} and not empty.
-     * Otherwise returns an empty {@code List}.
-     */
-    protected List<String> getManyValueInElement(final String[] elements, int position) {
-        if (!containsValueAtPosition(elements, position)) {
+        if (rawValue.isEmpty()) {
             return Collections.emptyList();
         }
+        return Arrays.stream(rawValue.split((COLUMN_VALUES_SEPARATOR))).map(String::trim).collect(Collectors.toList());
```

Comment:
Can use `getAsList`.

Reply:
I don't really know how to do that here. the `rawValue` is `String` and not `CSVRecord`, so I can't just call `getAsList` because of the override keyword.

Reply:
🤔 Hmm, seems like its not so easy and direct.

There are methods that rely on `get` and its method signature `(final CSVRecord record, int colNum)`.

Let me think about it first. **Don't do anything drastic yet**.


```
protected List<String> getAsList(final CSVRecord record, int colNum) {
>    if (get(record, colNum).isEmpty()) {
>        return Collections.emptyList();
>    }
>    return Arrays.stream(get(record, colNum).split(COLUMN_VALUES_SEPARATOR))
>            .map(String::trim)
>            .collect(Collectors.toList());
}

protected List<String> getAsListWithoutOverridePrefix(final CSVRecord record, int colNum) {
    String rawValue = isElementOverridingStandaloneConfig(record, colNum)
            ? get(record, colNum).replaceFirst(OVERRIDE_KEYWORD, "")
            : get(record, colNum);

>    if (rawValue.isEmpty()) {
>        return Collections.emptyList();
>    }
>    return Arrays.stream(rawValue.split((COLUMN_VALUES_SEPARATOR)))
>		    .map(String::trim)
>		    collect(Collectors.toList());
}
```

Reply:
@jylee-git 

You can extract the similar code chunks/SLAP first.

Reply:
```java
protected List<String> getAsListWithoutOverridePrefix(final CSVRecord record, int colNum) {
    List<String> data = getAsList(record, colNum);
    if (isElementOverridingStandaloneConfig(record, colNum)) {
        data.set(0, data.get(0).replaceFirst(OVERRIDE_KEYWORD, ""));
    }
    return data;
}
```

?

Reply:
Nice! 👍